src/pkg/webhook/builder.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2022 Amazon.com, Inc. or its affiliates. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	urlpkg "net/url"
+	templatepkg "text/template"
+	"time"
+)
+
+type (
+	HttpSendFunc = func(*http.Request) (*http.Response, error)
+	ProxyFunc    = func(*http.Request) (*url.URL, error)
+
+	ClientBuilder func(ProxyFunc) HttpSendFunc
+)
+
+func NewHttpClientDo(proxy ProxyFunc) HttpSendFunc {
+	c := &http.Client{
```

Comment:
Does the HTTP client need any retry or backoff configurations, or is this sufficient for how it makes requests?

Reply:
This is how NTHv1 configures its client (see [here](https://github.com/aws/aws-node-termination-handler/blob/5e4984f2d8588ce6661cc33ccb836249a85852a8/pkg/webhook/webhook.go#L93-L108)).  Might be worth adding retry, etc. support if we run into problems.